illumio_connector.py

```diff
@@ -1,249 +1,341 @@
-        action = self.get_action_identifier()
-        self.debug_print("action_id", action)
-
-        action_mapping = {
-            'test_connectivity': self._handle_test_connectivity,
-            'create_virtual_service': self._handle_create_virtual_service,
-            # 'traffic_analysis': self._handle_traffic_analysis,
-            # 'create_service_binding': self._handle_service_binding,
-            # 'create_ruleset': self._handle_create_ruleset,
-            # 'create_rule': self._handle_create_rule,
-            # 'create_enforcement_boundary': self._handle_create_enforcement_boundary,
-            # 'provision_objects': self._handle_provision_objects,
-            # 'get_workload': self._handle_get_workload,
-            # 'get_workloads': self._handle_get_workloads,
-            # 'get_ip_list': self._handle_get_ip_list,
-            # 'update_enforcement_mode': self._handle_update_enforcement_mode
-        }
+        except IllumioException as e:
+            self.set_status(
+                phantom.APP_ERROR, f"Encountered error while conecting to PCE: {e}"
+            )
+            return None
 
-        status = phantom.APP_SUCCESS
+        return pce if test_connection else None
 
-        if action in action_mapping:
-            action_function = action_mapping[action]
-            status = action_function(param)
+    def convert_object_to_json(self, obj):
+        """
+        Converts result object to json
 
-        return status
+        :param obj: result object
+        :return : json
+        """
 
-    def initialize(self):
-        # Load the state in initialize, use it to store data
-        # that needs to be accessed across actions
-        self._state = self.load_state()
+        try:
+            json_data = json.loads(json.dumps(obj, cls=illumio.util.IllumioEncoder))
+        except IllumioException as e:
+            self.set_status(
+                phantom.APP_ERROR, f"Encountered error while processing response: {e}"
+            )
+            return {}
+        return json_data
+
+    def _validate_integers(self, action_result, parameter, key, allow_zero=False):
+        """
+        Checks if the provided input parameter value
+        is a non-zero positive integer and returns the integer value of the parameter itself.
+
+        :param action_result: Action result or BaseConnector object
+        :param parameter: input parameter
+        :param key: input parameter message key
+        :param allow_zero: whether zero should be considered valid value or not
+        :return: integer value of the parameter or None in case of failure
+        """
+
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    action_result.set_status(
+                        phantom.APP_ERROR,
+                        f"Please provide a valid integer value in the {key} parameter",
+                    )
+                    return None
```

Comment:
Returning None here confuses the implementation - if we're setting the result status this is unnecessary. If we must return early, just return `parameter`.

Reply:
Updated code